assets/js/googlepay.js

```diff
@@ -0,0 +1,54 @@
+(function ($) {
+    'use strict';
+
+    $(document).ready(function() {
+        setTimeout(function() {
+            checkGooglePaySupport();
+        }, 100);
+    });
+
+    $(document).on("DOMNodeInserted", '.woocommerce-checkout-payment', function(e) {
+        checkGooglePaySupport();
+    });
+
+    /**
+     * Check the deive to support google pay.
+     */
+    function checkGooglePaySupport() {
+        $('label[for$=payrexx_google-pay]').hide();
+        try {
+            const baseRequest = {
+                apiVersion: 2,
+                apiVersionMinor: 0
+            };
+            const allowedCardNetworks = ['MASTERCARD', 'VISA'];
+            const allowedCardAuthMethods = ['CRYPTOGRAM_3DS'];
+            const baseCardPaymentMethod = {
+                type: 'CARD',
+                parameters: {
+                    allowedAuthMethods: allowedCardAuthMethods,
+                    allowedCardNetworks: allowedCardNetworks
+                }
+            };
+
+            const isReadyToPayRequest = Object.assign({}, baseRequest);
+            isReadyToPayRequest.allowedPaymentMethods = [
+                baseCardPaymentMethod
+            ];
+            const paymentsClient = new google.payments.api.PaymentsClient(
+                {
+                    environment: 'TEST'
```

Comment:
@vinothss4u  What does environment mean? Should this always be 'TEST'?

Reply:
@jdluca22 Yes, It is 'TEST'. In the Plugin, just check google support or not.